src/Controller/Traits/UserValidationTrait.php

```diff
@@ -40,6 +41,9 @@ public function validate($type = null, $token = null)
                     try {
                         $result = $this->getUsersTable()->validate($token, 'activateUser');
                         if ($result) {
+                            $event = $this->dispatchEvent(UsersAuthComponent::EVENT_AFTER_REGISTER_CONFIRM, [
```

Comment:
once the event is dispatched, we should check for data, and possibly redirect there, as we do on similar events. What do you think?

Reply:
I tried first to put make the eventManager on activate function but there is a redirect to login page.